src/error.c

```diff
@@ -15,97 +15,102 @@
     ECode(WrongDoubleQuote, "User entered `“` (0x201c) as a quote instead of ASCII"), \
     ECode(Syntax, "Syntax error"), \
     ECode(Fatal, "Fatal error"), \
+    ECode(Redefinition, "Redefinition"), \
 
 typedef enum ErrorCode {
 #define ECode(e, s) e##Error
     ERROR_CODES
 #undef ECode
 } ErrorCode;
 
-typedef struct Error {
-    ErrorCode code;
-    Position pos;
-    const char *message;
-} Error;
-
-typedef struct ErrorCollector {
-    u32 errorCount;
-} ErrorCollector;
+typedef struct DiagnosticNote DiagnosticNote;
+struct DiagnosticNote {
+    const char *msg;
+    DiagnosticNote *next;
+};
 
-ErrorCollector errorCollector;
+typedef struct DiagnosticError DiagnosticError;
+struct DiagnosticError {
+    const char *msg;
+    DiagnosticNote *note;
+};
 
-b32 HasErrors() {
-    return errorCollector.errorCount > 0;
+b32 shouldPrintErrorCode() {
+#if NO_ERROR_CODES
+    return false;
+#endif
+    return FlagErrorCodes;
 }
 
-#define tempErrorBufferLen KB(4)
-char *temporaryErrorBuffer;
+void ReportError(Package *p, ErrorCode code, Position pos, const char *msg, ...) {
+    va_list args;
+    char msgBuffer[512]; // TODO: Static & Thread Local?
```

Comment:
Just a note, in the future we may have translated errors (I think we'll keep translations purely in the kai-man tool). And if so this isn't very many characters for asian languages that take 3 bytes/character

Reply:
Yeah I looked up the default stack size on Windows and got scared (1MB). That's the reason for the todo for thread local